### PR TITLE
Add complete 2.2 compatibility

### PR DESCRIPTION
**Pull request by lewiscollard**
Since I am rewriting the docs (#185) and the docs say it should be compatible with the latest LTS version of Django, I figure I should work backwards and make the code match the docs (but really it's overdue anyway).

Reading:

[Django 1.x to 2.x incompatibilities](https://docs.djangoproject.com/en/2.0/releases/2.0/#deprecated-features-2-0)

[Deprecated in 2.2](https://docs.djangoproject.com/en/2.2/releases/2.2/#features-deprecated-in-2-2)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -33,6 +33,12 @@ jobs:
           command: |
             . .venv/bin/activate
             pytest --cov=cms -n auto --create-db --nomigrations
+      - run:
+          name: pytest-1.11
+          command: |
+            . .venv/bin/activate
+            pip install 'django>=1.11,<1.12'
```
Comment (daniel-clayton) [inline .circleci/config.yml]: Should this not be using 2.2?

File: cms/apps/media/tests/test_file_used_on.py [hunk 2]
```diff
@@ -27,33 +28,43 @@ class TestModelBase(models.Model):
...
     pass
 
+
 class TestContentBase(TestModelBase, ContentBase):
     page = models.OneToOneField(
         Page,
         primary_key=True,
         editable=False,
         related_name='+',
+        on_delete=models.CASCADE
```
Comment (daniel-clayton) [inline cms/apps/media/tests/test_file_used_on.py]: Trailing comma pls :)

File: cms/tests/test_forms.py [hunk 3]
```diff
@@ -29,13 +30,22 @@ def test_htmlwidget_get_media(self):
...
-                '/static/cms/js/tinymce/tinymce.min.js',
-                '/static/cms/js/jquery.cms.wysiwyg.js'
-            ]
-        })
+        if (VERSION[0], VERSION[1]) == (1, 11):
+            self.assertDictEqual(media.__dict__, {
+                '_css': {},
+                '_js': [
+                    '/static/cms/js/tinymce/tinymce.min.js',
+                    '/static/cms/js/jquery.cms.wysiwyg.js'
```
Comment (daniel-clayton) [inline cms/tests/test_forms.py]: ,

File: cms/tests/test_forms.py [hunk 4]
```diff
@@ -29,13 +30,22 @@ def test_htmlwidget_get_media(self):
...
+                ],
+            })
+        else:
+            self.assertDictEqual(media.__dict__, {
+                '_css_lists': [{}],
+                '_js_lists': [[
+                    '/static/cms/js/tinymce/tinymce.min.js',
+                    '/static/cms/js/jquery.cms.wysiwyg.js'
```
Comment (daniel-clayton) [inline cms/tests/test_forms.py]: ,

File: .circleci/config.yml (hunk 1 above)
Comment (lewiscollard) [reply .circleci/config.yml]: The other pytest block above it tests for 2.2 (which is automatically installed by setup.py). This is a second run of tests to ensure that it still works under 1.11.

**Pull request merged**